terraform/cloudfront/cloudfront_distribution/distribution.tf

```diff
@@ -0,0 +1,61 @@
+resource "aws_route53_record" "cdn_domain" {
+  count = "${length(var.aliases)}"
+
+  zone_id = "${var.system_dns_zone_id}"
+  name    = "${element(var.aliases, count.index)}."
+  type    = "CNAME"
+  ttl     = "60"
+  records = ["${aws_cloudfront_distribution.cdn_instance.domain_name}"]
+}
+
+resource "aws_cloudfront_distribution" "cdn_instance" {
+  origin {
+    domain_name = "${var.origin}"
+    origin_id   = "${var.origin}"
+
+    custom_origin_config {
+      origin_protocol_policy = "https-only"
+      http_port              = 80
+      https_port             = 443
+      origin_ssl_protocols   = ["TLSv1", "TLSv1.1", "TLSv1.2"]
+    }
+  }
+
+  aliases         = "${var.aliases}"
+  comment         = "${var.comment}"
+  enabled         = true
+  is_ipv6_enabled = true
+
+  default_cache_behavior {
+    allowed_methods  = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = "${var.origin}"
+
+    forwarded_values {
+      query_string = true
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    viewer_protocol_policy = "redirect-to-https"
+    min_ttl                = 0
+    default_ttl            = 3600
+    max_ttl                = 86400
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+
+  tags {
+    Environment = "${var.env}"
+  }
+
+  viewer_certificate {
+    cloudfront_default_certificate = true
```

Comment:
Do we need to use the wildcard certs that are uploaded by [`make <env> manually_upload_certs`](https://github.com/alphagov/paas-cf/blob/3e3ba8b18d2c9ccb6928713ad4e9e0859b0e92e5/Makefile#L185-L192) in order for the client browser to get a valid cert when you visit `www.cloud.service.gov.uk` or `docs.cloud.service.gov.uk`?

Not sure how you'd test it in dev, but I think we should mention it in the review section of the PR.